File: backend/src/main/java/vkr/backend/services/MessageService.java
package vkr.backend.services;

import vkr.backend.entities.Message;

import java.util.List;

public interface MessageService {

    /**
     * Отправляет сообщение в указанном диалоге
     *
     * @param dialogId    id диалога, в котором отправляется сообщение
     * @param senderId    id отправителя сообщения
     * @param recipientId id получателя сообщения
     * @param text        Текст сообщения
     * @return Отправленное сообщение
     * @throws IllegalArgumentException если диалог или пользователи не найдены
     */
    Message sendMessage(Long dialogId, Long senderId, Long recipientId, String text);

    /**
     * Редактирует текст сообщения
     *
     * @param messageId id сообщения, которое нужно отредактировать
     * @param newText   Новый текст сообщения
     * @return Обновленное сообщение
     * @throws IllegalArgumentException если сообщение не найдено
     */
    Message editMessage(Long messageId, String newText);

    /**
     * Удаляет сообщение по его id
     *
     * @param messageId id сообщения, которое нужно удалить
     * @throws IllegalArgumentException если сообщение не найдено
     */
    void deleteMessage(Long messageId);

    /**
     * Получает все сообщения в указанном диалоге
     *
     * @param dialogId id диалога, сообщения которого нужно получить
     * @return Список сообщений в диалоге
     */
    List<Message> getMessagesByDialogId(Long dialogId);
}

File: frontend/src/services/messageService.js
import axios from 'axios';
import authHeader from './authHeader';

const API_URL = '/api/messages';

class MessageService {
    sendMessage(dialogId, senderId, recipientId, text) {
        return axios.post(`${API_URL}/send`, null, {
            params: {dialogId, senderId, recipientId, text},
            headers: authHeader()
        }).then(response => response.data)
            .catch(error => {
                const _content = (error.response && error.response.data) || error.message || error.toString();
                console.error("Ошибка при отправке сообщения:", _content);
                throw error;
            });
    }

    editMessage(id, newText) {
        return axios.put(`${API_URL}/edit/${id}`, null, {
            params: {newText},
            headers: authHeader()
        }).then(response => response.data)
            .catch(error => {
                const _content = (error.response && error.response.data) || error.message || error.toString();
                console.error("Ошибка при редактировании сообщения:", _content);
                throw error;
            });
    }

    deleteMessage(id) {
        return axios.delete(`${API_URL}/delete/${id}`, {headers: authHeader()})
            .then(response => response.data)
            .catch(error => {
                const _content = (error.response && error.response.data) || error.message || error.toString();
                console.error("Ошибка при удалении сообщения:", _content);
                throw error;
            });
    }

    getMessagesByDialogId(dialogId) {
        return axios.get(`${API_URL}/dialog/${dialogId}`, {headers: authHeader()})
            .then(response => response.data)
            .catch(error => {
                const _content = (error.response && error.response.data) || error.message || error.toString();
                console.error("Ошибка при получении сообщений:", _content);
                throw error;
            });
    }
}

export default new MessageService();
